### DISCUSSION TITLE
Copy directory recursively?

### DISCUSSION THREAD

**Discussion by Dieterbe** [CyberShadow/aconfmgr#164] [Q&A]
I have a lot of entries like this (one per wifi network i've ever connected to)
```
CopyFile /etc/NetworkManager/system-connections/blah.nmconnection 600
```
and every few weeks when i run 'aconfmgr save' i usually have a few more.

my general philosphy on this is, i rather not ignore these, and prefer to include them all, so that a restored system can immediately access any network i've previously connected to.
However i don't care about all the individual files, i just see it as one ever growing set

Is there a way to just have 1 directive that says "copy the entire directory" (both on apply and save) ? Seems CopyFile and CopyFileTo only work with files, not directories?

i haven't tried it yet, but maybe the solution is a helper similar to CopyFileTo, that uses `cp -a` and updates `used_files` for all files within? i'm probably missing some stuff.

**Comment by CyberShadow** [comment]
Making `CopyFile` recursive is half of it; another half would be to make `aconfmgr save` recognize that the configuration file is copying some directory recursively, so it doesn't need to emit `CopyFile` directives for missing files. However... that does go against aconfmgr's philosophy a bit, in that any change between the system and configuration is described by a line added to `99-unsorted.sh`.

For now I suggest to approach this pragmatically and use a script which runs after `aconfmgr save` and automatically moves `CopyFile` directives to their right place, or copying the directory recursively then making the script delete unnecessary `CopyFile` directives.

**Comment by Dieterbe** [comment]
>For now I suggest to approach this pedantically and use a script which runs after aconfmgr save and automatically moves CopyFile directives to their right place

fair enough, i'll do this. thanks
